src/embedding_client.py

```
@@ -214,43 +229,64 @@ def _create_batches(
+                        )
+
+                return dict(result)
+
+            except Exception as e:
+                last_exception = e
+                if attempt < max_retries - 1:
+                    # Exponential backoff: 1s, 2s, 4s
+                    wait_time = 2**attempt
+                    logger.warning(
+                        f"Embedding batch failed (attempt {attempt + 1}/{max_retries}), "
+                        + f"retrying in {wait_time}s: {e}"
```

Ruff (0.14.10)

Explicitly concatenated string should be implicitly concatenated

Remove redundant '+' operator to implicitly concatenate

(ISC003)